MED-997

Cymek12 opened the pull request:
+ H2Database
+ JPA
+ PatientDAO

Cymek12 committed:
+ PatientDAO
+ H2Database
+ JPA

Cymek12 committed:
+ PatientDAO
+ H2Database
+ JPA

Cymek12 committed:
+ odpowiednia zmiana metod, aby JPARepisotry działało poprawnie

Cymek12 committed:
+ odpowiednia zmiana metod, aby JPARepisotry działało poprawnie

Cymek12 committed:
poprawki PatientService po testach

Cymek12 committed:
+ Doctor
+ Institution

Cymek12 committed:
+ Institution
+ Doctor

konradgab commented on src/main/java/com/example/demo/controller/DoctorController.java:
```diff
@@ -0,0 +1,48 @@
+        doctorService.deleteDoctorByEmail(email);
+    }
+
+    @PatchMapping("/addInstitution/{email}/{id}")
```
Starajmy sie w sciezkach nie uzywac czasownikow najprosciej mowiac. przykladowe wywolanie 
Localhost:8080/addInstitution/kg@gmail.com/1 <-  jest trochę nieoczywiste czym jest to 1 zastanow sie czy mozna to zrobic lepiej. Ogolnie raczej standardem w sciezkach do oddzielania od siebie wyrazów jest nie camel case a "-". Czyli
/add-intsitution/... jak juz

konradgab commented on src/main/java/com/example/demo/controller/DoctorController.java:
```diff
@@ -0,0 +1,48 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/doctor")
```
liczba mnoga raczej -> operujesz na doktorach -> /doctors

konradgab commented on src/main/java/com/example/demo/controller/InstitutionController.java:
```diff
@@ -0,0 +1,43 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/institution")
```
liczba mnoga raczej -> operujesz na institutions-> /institutions

konradgab commented on src/main/java/com/example/demo/controller/PatientController.java:
```diff
@@ -11,7 +11,7 @@
 
 @RestController
-@RequestMapping("/patients")
+@RequestMapping("/patient")
```
liczba mnoga raczej -> operujesz na patients-> /patients

konradgab commented on src/main/java/com/example/demo/service/DoctorService.java:
```diff
@@ -0,0 +1,97 @@
+        }
+    }
+
+    private void editDoctorData(Doctor doctor, Doctor newDoctorData) {
```
tą metodę może lepiej wynieść do klasy Doctor

konradgab commented on src/main/java/com/example/demo/service/DoctorService.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    private boolean isDoctorDataNull(Doctor doctor) {
+        return doctor.getEmail() == null ||
```
tą metodę może lepiej wynieść do klasy Doctor

konradgab commented on src/main/java/com/example/demo/service/DoctorService.java:
```diff
@@ -0,0 +1,97 @@
+                .orElseThrow(() -> new InstitutionNotFoundException("Instituion with id: " + id + " do not exists"));
+        doctor.getInstitutions().add(institution);
+        doctorRepository.save(doctor);
+
```
pusta linia

konradgab commented on src/main/java/com/example/demo/service/InstitutionService.java:
```diff
@@ -0,0 +1,80 @@
+        }
+    }
+
+    private void editInstitutionData(Institution institution, Institution newInstitutionData) {
```
moze lepiej wyniescc do institution

konradgab commented on src/main/java/com/example/demo/service/InstitutionService.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    private boolean isInstitutionDataNull(Institution institution) {
+        return institution.getName() == null ||
```
moze lepiej wyniescc do institution

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -18,33 +16,77 @@ public class PatientService {
+    }
+
+    private void editPatientData(Patient patient, Patient newPatientData) {
+        patient.setEmail(newPatientData.getEmail());
```
moze lepiej wyniesc do klasy Patient

konradgab commented on src/main/java/com/example/demo/service/PatientService.java:
```diff
@@ -18,33 +16,77 @@ public class PatientService {
+        patient.setBirthday(newPatientData.getBirthday());
+    }
+
+    private boolean isPatientDataNull(Patient patient) {
```
moze lepiej wyniesc do klasy Patient

Cymek12 committed:
+ Institution
+ Doctor

Cymek12 committed:
+ equals and hashcode

Cymek12 committed:
+ Pageable
+ CascadeType

Cymek12 committed:
+ Pageable
+ CascadeType

Cymek12 committed:
+ Pageable
+ CascadeType

Cymek12 committed:
+ Pageable
+ CascadeType

konradgab commented on src/main/java/com/example/demo/controller/DoctorController.java:
```diff
@@ -0,0 +1,48 @@
+        doctorService.deleteDoctorByEmail(email);
+    }
+
+    @PatchMapping("/add-institution/{doctorEmail}/{institutionId}")
```
czasownik w sciezce

konradgab commented on src/main/java/com/example/demo/model/entity/Institution.java:
```diff
@@ -0,0 +1,58 @@
+    @ManyToMany(mappedBy = "institutions", cascade = {CascadeType.PERSIST, CascadeType.MERGE})
+    private Set<Doctor> doctors;
+
+
```
nadmiarowa pusta linia

konradgab commented on src/main/java/com/example/demo/model/entity/Institution.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    public boolean isInstitutionDataNull() {
+        return this.getName() == null ||
```
sprobuj se isNull uzyc

konradgab commented on src/main/java/com/example/demo/model/entity/Patient.java:
```diff
@@ -0,0 +1,61 @@
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
```
ja bym chyba zrezygnowal z Data

konradgab commented on src/main/java/com/example/demo/model/entity/Institution.java:
```diff
@@ -0,0 +1,58 @@
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
```
zapętlenie dla metody toString

konradgab commented on src/main/java/com/example/demo/service/InstitutionService.java:
```diff
@@ -0,0 +1,142 @@
+
+    public void addInstitutionsWithDoctors(List<FullInstitutionDTO> fullInstitutionDTOS) {
+        Set<Institution> institutions = fullInstitutionDTOS.stream()
+                .map(fullInstitutionDTO -> {
```
side effect w streamie raczej unikaj takich rzeczy zastanow sie jak inaczej to napisac

Cymek12 committed:
change post mapping path

Cymek12 committed:
fix bugs

Cymek12 committed:
+ Pageable
+ CascadeType

Cymek12 committed:
Merge remote-tracking branch 'origin/feature/MED-997' into feature/MED-997

Cymek12 committed:
+ tests

Cymek12 committed:
PatientService tests

Cymek12 committed:
+ PatientServiceTest.java

Cymek12 committed:
+ DoctorServiceTest.java
+ PatientServiceTest.java

konradgab commented on pom.xml:
```diff
@@ -56,6 +56,21 @@
+		</dependency>
+		<dependency>
+			<groupId>org.junit.jupiter</groupId>
+			<artifactId>junit-jupiter-engine</artifactId>
```
po co to? Spring boot starter tests zawiera juz junit

konradgab commented on src/test/java/com/example/demo/service/DoctorServiceTest.java:
```diff
@@ -0,0 +1,326 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+
+import javax.print.Doc;
```
po co to?

konradgab commented on src/test/java/com/example/demo/service/DoctorServiceTest.java:
```diff
@@ -0,0 +1,326 @@
+        Pageable pageable = PageRequest.of(0, 5);
+        Doctor doctor = buildDoctor();
+        List<Doctor> doctors = new ArrayList<>();
+        doctors.add(doctor);
```
moglibysmy użyć List.of albo Collections.singletonList()

konradgab commented on src/test/java/com/example/demo/service/DoctorServiceTest.java:
```diff
@@ -0,0 +1,326 @@
+    }
+
+    @Test
+    void getDoctors_doctorsExists_returnPageContentDTO() {
```
Testując listę lepiej jest sprawdzić 2 elementy

konradgab commented on src/test/java/com/example/demo/service/DoctorServiceTest.java:
```diff
@@ -0,0 +1,326 @@
+    }
+
+    @Test
+    void addInstitution_doctorExistsInstitutionExists_addInstitutionToDoctor() {
```
addInstitution_DataCorrect_InstitutionAssigned

konradgab commented on src/test/java/com/example/demo/service/DoctorServiceTest.java:
```diff
@@ -0,0 +1,326 @@
+        //when
+        DoctorDTO resultDoctorDTO = doctorService.addInstitution(email, id);
+        //then
+        assertEquals(1L, resultDoctorDTO.getInstitutionIds().getFirst());
```
mozna sie pokusic o wiecej asercji -> to jest przypisanie to moze sprawdzmy czy przypisana instytucja zawiera dobre dane

konradgab commented on src/test/java/com/example/demo/service/PatientServiceTest.java:
```diff
@@ -0,0 +1,351 @@
+    }
+
+    @Test
+    void addPatient_patientDataIsNull_exceptionThrown() {
```
addPatient_patientDataIsNull_PatientDataIsNullExceptionThrownn

konradgab commented on src/test/java/com/example/demo/service/PatientServiceTest.java:
```diff
@@ -0,0 +1,351 @@
+        PatientDTO result = patientService.editPassword(email, passwordRequest);
+        //then
+        assertEquals(1L, result.getId());
+        assertEquals("test@gmail.com", result.getEmail());
```
mozesz sprawdzic przy uzyciu mockito.verify czy do metody save został przekazany pacjent z okreslonym nowym haslem -> ze wzgledu na implementacje metody equals dla encji to musimy napoiisac aby to pokryc odpowiedi argument matcher albo uzyc argument captor

Cymek12 committed:
+ DoctorServiceTest.java
+ PatientServiceTest.java
+ InstitutionServiceTest.java
+ VisitServiceTest.java

Cymek12 committed:
+ DoctorServiceTest.java
+ PatientServiceTest.java
+ InstitutionServiceTest.java
+ VisitServiceTest.java

Cymek12 committed:
+ DoctorServiceTest.java
+ PatientServiceTest.java
+ InstitutionServiceTest.java
+ VisitServiceTest.java

Cymek12 committed:
dodanie zaleznosci liquibase
+ master.yaml
+ MEDICAL_CLINIC.yaml

Cymek12 committed:
+ zmiana nazewnictwa kolumn w plikach yaml

Cymek12 committed:
uzupelnienie brakow w testach controllerow

Cymek12 committed:
uzupelnienie brakow w testach controllerow

Cymek12 committed:
uzupelnienie brakow

Cymek12 committed:
OpenFeign

Cymek12 committed:
OpenFeign

konradgab commented on pom.xml:
```diff
@@ -26,38 +29,54 @@
+
 	<properties>
-		<java.version>21</java.version>
+		<java.version>23</java.version>
```
co ty robisz

konradgab commented on pom.xml:
```diff
@@ -26,38 +29,54 @@
+			<artifactId>spring-boot-starter-data-jpa</artifactId>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework.data</groupId>
```
to chyba niepotrzebne bo mamy spring-boot-starter-data-jpa

konradgab commented on pom.xml:
```diff
@@ -26,38 +29,54 @@
 		</dependency>
 		<dependency>
-			<groupId>org.mapstruct</groupId>
-			<artifactId>mapstruct-processor</artifactId>
```
ta biblioteka powinna byc ponieważ ona wlasnie generuje mappery mapstructowe -> zweryfikuj czy masz tą biblioteke bo jest ona potrzebna

konradgab commented on src/main/java/com/example/demo/controller/DoctorController.java:
```diff
@@ -0,0 +1,47 @@
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.*;
+
```
nadmiarowa pusta linia

konradgab commented on src/main/java/com/example/demo/controller/DoctorController.java:
```diff
@@ -0,0 +1,47 @@
+        doctorService.deleteDoctorByEmail(email);
+    }
+
+    @PatchMapping("/institution/{doctorEmail}/{institutionId}")
```
sciezki do poprawy tak jak rozmawialismy

konradgab commented on src/main/java/com/example/demo/controller/VisitController.java:
```diff
@@ -0,0 +1,47 @@
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.*;
+
```
puste

konradgab commented on src/main/java/com/example/demo/controller/VisitController.java:
```diff
@@ -0,0 +1,47 @@
+        return visitService.getVisits(pageable);
+    }
+
+    @GetMapping("/{patientEmail}")
```
sciezka do zmiany

Cymek12 committed:
OpenFeign

Cymek12 committed:
OpenFeign

Cymek12 committed:
poprawa testów
+ swagger
